### DISCUSSION TITLE
What message delivery guarantees are provided by uWebSockets.js?

### DISCUSSION THREAD

**Discussion by Eric24** [uNetworking/uWebSockets.js#1020] [Q&A] [answered]
- Guaranteed delivery (or error)?
- Guaranteed "one and only one" message (i.e. no duplicates)?
- Any others?

**Comment by uNetworkingAB** [comment, marked as answer]
This lib only implements standard WebSocket so there is no particular delivery guarantee other than what you get with standard WebSocket (which is TCP). You can implement some application level ACK on top if you want that but then you can't really communicate with any client
